### DISCUSSION TITLE
Make `value` from Segment's onChange() non-optional

### DISCUSSION THREAD

**Discussion by seanlaff** [grafana/grafana#36975] [General]
In Segment and SegementAsync, the onChange function returns an object with an optional `value` attribute:

```ts
export interface SelectableValue<T = any> {
    label?: string;
    value?: T;
    imgUrl?: string;
    description?: string;
    [key: string]: any;
}
```
Could `value` instead be non-optional? It makes using the component a little painful as is since I always need to null check, even if I don't provide a nullable type to the generic.

I have to do 
```ts
<SegmentAsync<string> onChange={v => v.value && this.handleAddGroupBy(v.value)} loadOptions={this.loadFieldOptions} />
```

If I wanted undefined to be a valid option, I would rather do it explicitly with the generic type, ala
```ts
<SegmentAsync<string | undefined> onChange={v => v.value && this.handleAddGroupBy(v.value)} loadOptions={this.loadFieldOptions} />
```

Thoughts?

**Comment by dprokop** [comment]
@seanlaff yes, I think this would make sense to be changed. This will require some refactoring though, as naively changing the type to non-optional is causing quite a lot of TS errors.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
